### PR TITLE
Allow saving multiple media items in conversation

### LINKED ISSUE #5874: Allow saving multiple images in a conversation
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Feature Request
There was #5088 asking for multi-selecting images in "All images" and then saving them. It was closed without actually having multi-select & save implemented.
I am asking for multi-select images in a conversation and then save them.

### Steps to reproduce
- go to a conversation
- long press a picture message
- tap a second/third/... picture message

**Actual result:** No save button is shown.
**Expected result:** A save button should be shown and allow to save the selected images.

### PR DESCRIPTION
**Pull request by FeuRenard**
### Contributor checklist

- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
 * Sony Xperia U, Android 4.4.4 (CyanogenMod)
 * AVD Nexus 5X, Android 6.0
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in the commit message of my first commit

----------

### Description
This PR will enable saving multiple media attachments at once, if the user selects them in a conversation.

Fixes #5874
// FREEBIE

### CODE REVIEW

File: src/org/thoughtcrime/securesms/ConversationFragment.java [hunk 1]
```diff
@@ -354,23 +356,46 @@ protected Void doInBackground(MessageRecord... messageRecords) {
...
+          @Override
+          protected List<Attachment> doInBackground(Void... params) {
+            List<Attachment> attachments = new ArrayList<>(messages.size());
+            for (MediaMmsMessageRecord mediaMmsMessageRecord : messages) {
+              for (Slide slide : mediaMmsMessageRecord.getSlideDeck().getSlides()) {
+                if (slide.hasImage() || slide.hasVideo() || slide.hasAudio()) {
+                  attachments.add(new Attachment(slide.getUri(),
+                                                 slide.getContentType(),
+                                                 mediaMmsMessageRecord.getDateReceived()));
+                }
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/ConversationFragment.java]: why is this an async task?

Comment (FeuRenard) [inline src/org/thoughtcrime/securesms/ConversationFragment.java]: I just adapted it from here without further scrutinizing it:
https://github.com/WhisperSystems/Signal-Android/blob/0a569676f7a57144374a24faef566b2ca3233290/src/org/thoughtcrime/securesms/MediaOverviewActivity.java#L149

Comment (FeuRenard) [inline src/org/thoughtcrime/securesms/ConversationFragment.java]: Ok, now I see the difference. In MediaOverviewActivity the database is accessed, therefore an async task. Should I change it here?

Comment (moxie0) [inline src/org/thoughtcrime/securesms/ConversationFragment.java]: Yes, you only need to use an async task if you're doing IO from the UI thread

**Comment by FeuRenard** [comment]
Now the attachments are fetched synchronously. Some smaller code polishing to follow.

**Comment by FeuRenard** [comment]
I rebased this PR. Due to changes to the way the save success is shown to the user there is now a toast which says "Saved to <directory>". In the case of multiple media <directory> == null and so says the toast: "Saved to null".

**Pull request closed without merging**